pkg/pillar/types/types.go

```diff
@@ -48,6 +48,12 @@ const (
 	PURGING    // Purging due to config change
 	BROKEN     // Domain is still alive, but its device model has failed
 	UNKNOWN    // State of the domain can't be determined
+	// PENDING to start
+	PENDING
+	// SCHEDULING waiting to be scheduled
+	SCHEDULING
+	// FAILED to start
+	FAILED
```

Comment:
Don't we need to define these in the protobuf enum in the API? If we want the controllers to see then that would make sense.

In any case, you also need to add these to
func (state SwState) ZSwState() info.ZSwState {
to avoid a log.Fatal for these new enum values.

Reply:
ok will do.

Reply:
PR submitted for eve-api https://github.com/lf-edge/eve-api/pull/55